neuroHarmonize/harmonizationLearn.py

```diff
@@ -101,53 +116,123 @@ def harmonizationLearn(data, covars, eb=True, smooth_terms=[],
+            gamma_star, delta_star = find_parametric_adjustments(s_data, LS_dict, info_dict)
+        else:
+            gamma_star = LS_dict['gamma_hat']
+            delta_star = np.array(LS_dict['delta_hat'])
+        bayes_data = adjust_data_final(s_data, design, gamma_star, delta_star, stand_mean, var_pooled, info_dict)
+        # save model parameters in single object
+        model = {'design': design, 'SITE_labels': batch_labels,
+                'var_pooled':var_pooled, 'B_hat':B_hat, 'grand_mean': grand_mean,
+                'gamma_star': gamma_star, 'delta_star': delta_star, 'info_dict': info_dict,
+                'gamma_hat': LS_dict['gamma_hat'], 'delta_hat': np.array(LS_dict['delta_hat']),
+                'gamma_bar': LS_dict['gamma_bar'], 't2': LS_dict['t2'],
+                'a_prior': LS_dict['a_prior'], 'b_prior': LS_dict['b_prior'],
+                'smooth_model': smooth_model, 'eb': eb,'SITE_labels_train':batch_labels,'Covariates':covar_levels}
+        # transpose data to return to original shape
+        bayes_data = bayes_data.T
     else:
-        gamma_star = LS_dict['gamma_hat']
-        delta_star = np.array(LS_dict['delta_hat'])
-    bayes_data = adjust_data_final(s_data, design, gamma_star, delta_star, stand_mean, var_pooled, info_dict)
-    # save model parameters in single object
-    model = {'design': design, 'SITE_labels': batch_labels,
-             'var_pooled':var_pooled, 'B_hat':B_hat, 'grand_mean': grand_mean,
-             'gamma_star': gamma_star, 'delta_star': delta_star, 'info_dict': info_dict,
-             'gamma_hat': LS_dict['gamma_hat'], 'delta_hat': np.array(LS_dict['delta_hat']),
-             'gamma_bar': LS_dict['gamma_bar'], 't2': LS_dict['t2'],
-             'a_prior': LS_dict['a_prior'], 'b_prior': LS_dict['b_prior'],
-             'smooth_model': smooth_model, 'eb': eb}
-    # transpose data to return to original shape
-    bayes_data = bayes_data.T
-    
+        # Create train data 
+        (batch_levels, sample_per_batch) = np.unique(covars[isTrainSite,batch_col],return_counts=True)
+        if batch_levels.size == 0:
+            bayes_data_train = np.zeros(shape=(0,data.shape[0]))
+            s_data_train = np.zeros(shape=(0,data.shape[0])).T
+        else:
+            info_dict_train = model['info_dict']
+            info_dict_train['sample_per_batch'] = sample_per_batch.astype('int')
+            info_dict_train['batch_info'] = [list(np.where(covars[isTrainSite,batch_col]==idx)[0]) for idx in batch_levels]
+            tmp = np.concatenate((np.zeros(shape=(info_dict['n_sample'],len(model['SITE_labels']))), design[:,len(batch_labels):]),axis=1)
+            s_data_train, stand_mean_train, _ = applyStandardizationAcrossFeatures(data[:,isTrainSite], tmp[isTrainSite,:], info_dict_train, model)
+            bayes_data_train = adjust_data_final(s_data_train, design[isTrainSite,:], model['gamma_star'], model['delta_star'], stand_mean_train, model['var_pooled'], info_dict_train)
```

Comment:
`design` is not correct here. It should contain a one-hot encoding of the `SITE` columns in the correct order.